### PR TITLE
[Stale][Feature] support partial update

### PR DESCRIPTION
**Pull request by xlfjcg**
Use enablePartialUpdate or sink.enable-partial-update to open

Signed-off-by: xlfjcg <1937532871@qq.com>

### CODE REVIEW

**Review by chaoyli — APPROVED** [review]

File: src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java [hunk 1]
```diff
@@ -201,10 +208,20 @@ public void enableUpsertDelete() {
         supportUpsertDelete = true;
     }
 
+    public void enablePartialUpdate() {
```
Comment (hffariel) [inline src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java:211]: no references of this method found.

**Review by hffariel — CHANGES_REQUESTED** [review]

File: src/main/java/com/starrocks/connector/flink/manager/StarRocksStreamLoadVisitor.java [hunk 2]
```diff
@@ -258,6 +257,9 @@ protected boolean isRedirectable(String method) {
                 }
                 httpPut.setHeader("columns", cols);
             }
+            if (sinkOptions.isEnablePartialUpdate()) {
```
[the remark this answers is not in the document]
Comment (xlfjcg) [reply src/main/java/com/starrocks/connector/flink/manager/StarRocksStreamLoadVisitor.java:260]: boolean field get method like isxxxx

File: src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java [hunk 3]
```diff
@@ -76,6 +76,8 @@ public enum StreamLoadFormat {
         .longType().defaultValue(600000L).withDescription("Offer to flushQueue timeout in millisecond.");
     public static final ConfigOption<Integer> SINK_METRIC_HISTOGRAM_WINDOW_SIZE = ConfigOptions.key("sink.metric.histogram-window-size")
         .intType().defaultValue(100).withDescription("Window size of histogram metrics.");
+    public static final ConfigOption<Boolean> SINK_ENABLE_PARTIAL_UPDATE = ConfigOptions.key("sink.enable-partial-update")
```
[the remark this answers is not in the document]
Comment (xlfjcg) [reply src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java:79]: partial update can just only set head in the moment, Here is the convenience for future expansion

File: src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java (hunk 1 above)
Comment (xlfjcg) [reply src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java:211]: custom job to can use

**Comment by waittttting** [comment]
LGTM

**Pull request closed without merging**